source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,992 @@
+            xReturn = eAzureIoTProvisioningInitFailed;
+        }
+        else
+        {
+            xReturn = eAzureIoTProvisioningSuccess;
+        }
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+void AzureIoTProvisioningClient_Deinit( AzureIoTProvisioningClientHandle_t xProvClientHandle )
+{
+    if( xProvClientHandle == NULL )
+    {
+        AZLogError( ( "AzureIoTProvisioningClient Deinit failed: Invalid argument" ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_SetSymmetricKey( AzureIoTProvisioningClientHandle_t xProvClientHandle,
+                                                                               const uint8_t * pucSymmetricKey,
+                                                                               uint32_t ulSymmetricKeyLength,
+                                                                               AzureIoTGetHMACFunc_t xHmacFunction )
+{
+    AzureIoTProvisioningClientResult_t xReturn;
+
+    if( ( xProvClientHandle == NULL ) ||
+        ( pucSymmetricKey == NULL ) || ( ulSymmetricKeyLength == 0 ) ||
+        ( xHmacFunction == NULL ) )
+    {
+        AZLogError( ( "AzureIoTProvisioning client symmetric key failed: Invalid argument" ) );
+        xReturn = eAzureIoTProvisioningInvalidArgument;
+    }
+    else
+    {
+        xProvClientHandle->_internal.pucSymmetricKey = pucSymmetricKey;
+        xProvClientHandle->_internal.ulSymmetricKeyLength = ulSymmetricKeyLength;
+        xProvClientHandle->_internal.pxTokenRefresh = prvProvClientGetToken;
+        xProvClientHandle->_internal.xHMACFunction = xHmacFunction;
+        xReturn = eAzureIoTProvisioningSuccess;
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_Register( AzureIoTProvisioningClientHandle_t xProvClientHandle,
+                                                                        uint32_t ulTimeoutMilliseconds )
+{
+    AzureIoTProvisioningClientResult_t xReturn;
+
+    if( xProvClientHandle == NULL )
+    {
+        AZLogError( ( "AzureIoTProvisioning registration failed: Invalid argument" ) );
+        xReturn = eAzureIoTProvisioningInvalidArgument;
+    }
+    else
+    {
+        if( xProvClientHandle->_internal.ulWorkflowState == azureiotprovisioningWF_STATE_INIT )
+        {
+            xProvClientHandle->_internal.ulWorkflowState = azureiotprovisioningWF_STATE_CONNECT;
+        }
+
+        xReturn = prvProvClientRunWorkflow( xProvClientHandle, ulTimeoutMilliseconds );
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_GetHub( AzureIoTProvisioningClientHandle_t xProvClientHandle,
```

Comment:
Should we call this AzureIoTProvisioningClient_GetDeviceAndHub, since it's returning both?